### ISSUE TITLE
Internal error using hollow function

### ISSUE THREAD

**Issue by rtsuk** [KittyCAD/modeling-app#4831]
> Modeling command failed: [ApiError { error_code: InternalEngine, message: "Invalid brep after shell operation" }]

![Screenshot](https://public.blob.zoo.dev/f0f1fafe-25f9-476f-a768-385b694506fc/modeling-app/coredump-8d36cda7-ce7e-4330-86e1-69ebcedc3505-screenshot.png)

<details>
<summary><b>Core Dump</b></summary>

[coredump-8d36cda7-ce7e-4330-86e1-69ebcedc3505.json](https://public.blob.zoo.dev/de6ba997-9cc4-4cdf-bc47-cf1e20db80b2/modeling-app/coredump-8d36cda7-ce7e-4330-86e1-69ebcedc3505.json)

Reference ID: 8d36cda7-ce7e-4330-86e1-69ebcedc3505
</details>

<details>
<summary><b>KCL Code</b></summary>

```kcl
coinDiameter = 50 // Diameter of the coin in mm
coinThickness = 2.0 // Thickness of the coin in mm
rimThickness = coinDiameter / 50 // Thickness of the rim around the edge in mm


exampleSketch = startSketchOn("XY")
  |> circle({
       center = [0, 0],
       radius = coinDiameter
     }, %)

example = hollow(1, extrude(coinThickness, exampleSketch))

```
</details>

**Comment by franknoirot** [comment]
Thanks for logging this @rtsuk! I'll pass this along to our engine team tomorrow

**Comment by franknoirot** [comment]
I think this is expected behavior, as you tried to use the `hollow` function with a wall thickness of `1` on a `2`mm tall item, leaving no remaining space to be made "hollow" after the shelling operation. Does that seem right @benjamaan476?

@rtsuk did you happen to use our Text-to-CAD feature to generate this? I'm interested in if it generated non-working code, which it shouldn't.

**Comment by franknoirot** [comment]
I'm going to close this as it is expected geometric behavior, but @rtsuk let me know when you get the chance if it was Text-to-CAD generated code here. And I'll just cc @KittyCAD/ml in case this is relevant to their continued improvements.

**Comment by benjamaan476** [comment]
You are correct @franknoirot, an exact value of 1 will mean there is no room to hollow, 0.99 works although the hollow inside would be tiny. If you meant to have a hollow that was 1 mm tall then you would use `hollow(0.5, extrude(coinThickness, exampleSketch))` as the first argument specifies the thickness of the walls after the hollow operation

**Comment by rtsuk** [comment]
I'm sorry to say I don't recall.

**Comment by greg-kcio** [comment]
I don't think the KCL here is generated by Text-to-CAD. If it was, you would see `// Generated by Text-to-CAD` at the top of the file.

**Comment by franknoirot** [comment]
No worries @rtsuk, thanks for getting back! We seem to have come to the conclusion that this is expected behavior of the `hollow` function, so I'm going to close this as resolved. Let me know if we can help you use Zoo in any other way!
